### ISSUE TITLE
[Bug] Current layer value changes automation not working

### ISSUE THREAD

**Issue by Alvie** [greghesp/ha-bambulab#775]
### Home Assistant Version

2024.11.3

### Bambu Lab Integration Version

v2.0.38

### Describe the bug

On creating an automation for current layer change, I am receiving an error message saying "Message malformed: must contain at least one of below, above:
![image](https://github.com/user-attachments/assets/1a3dc064-0229-484c-aa48-5e240e7651ec)

### To Reproduce

1. Create New Automation
2. Choose your Bambu Lab Printer as a trigger device.
3. Choose "Current layer value changes" for the when
4. Choose any action (e.g. Send notification)
5. Save

### Expected Behaviour

I expect the automation to save. Ideally whenever the layer number goes up, it triggers an event, without needing an above / below value.
[config_entry-bambu_lab-01JDBB762Y7GMWCJ8JKY843MP8.json](https://github.com/user-attachments/files/17878141/config_entry-bambu_lab-01JDBB762Y7GMWCJ8JKY843MP8.json)

### What device are you using?

A1 Mini

### Diagnostic Output

```shell
[config_entry-bambu_lab-01JDBB762Y7GMWCJ8JKY843MP8.json](https://github.com/user-attachments/files/17878141/config_entry-bambu_lab-01JDBB762Y7GMWCJ8JKY843MP8.json)
```

**Comment by AdrianGarside** [comment]
Home assistant is telling you that the automation you are creating requires an above or below value to be configured. This is unrelated to this integration.

**Comment by Alvie** [comment]
> Home assistant is telling you that the automation you are creating requires an above or below value to be configured. This is unrelated to this integration.

It seems that it is expecting there to be a value for above / below for the "current layer change" but there is no input to set this.

For the print length distance change, the inputs do appear.

![Screenshot_20241125-171311](https://github.com/user-attachments/assets/7e58256a-7932-4957-96d8-88236b6fd053)

Regardless, is there a way to trigger an event when the layer changes? I ask because I want to program the gcode to move the head out of the way (at the start of the layer), and eventually to get a different camera to take a snapshot so I can create a timelapse. Right now I just tried to get it to send a notification to my phone but I cannot even get it to do that.

Please let me know if this trigger is possible with this integration, thanks.

**Comment by AdrianGarside** [comment]
The integration must be indirectly influencing how home assistant is categorizing the sensor type. I’ll take a look.

**Comment by AdrianGarside** [comment]
But presumably you had to craft the automation in yaml since HA doesn't have the above file in the UI?

**Comment by jowlo** [comment]
> But presumably you had to craft the automation in yaml since HA doesn't have the above file in the UI?

No, i created it using the UI, you can either select the entity directly in a trigger (what i did) or first select the device and then choose the Layer Change event. 

I just pasted the yaml output here because you can copy-paste it better.

**Comment by AdrianGarside** [comment]
What version of home assistant / integration? I see the same as OP - no UI for above/below and an error when saving. I had to make a code change to set the unit type to a space character to get the UI to appear as it does for other measurement sensors.

**Comment by jowlo** [comment]
I am running HA `2024.11.2` and Bambulab Integration `v2.0.38`.

I did have some connection problems with newer firmware versions with Bambu Studio, so I downgraded my A1 Mini firmware to the one it came with. It was two versions older thant the current one, I don't know where to look it up (during a print). Is the firmware relevant here?

*PS: Added screenshots of the UI to the comment above*
